src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -1464,6 +1464,48 @@ namespace stan {
     }
     boost::phoenix::function<add_loop_identifier> add_loop_identifier_f;
 
+    void add_array_loop_identifier::operator()(const stan::lang::expression& expr,
+                                         std::string& name,
+                                         const scope& var_scope,
+                                         bool& pass, variable_map& vm,
+                                         std::stringstream& error_msgs) const {
+      std::size_t numdims = expr.expression_type().num_dims();
+      pass = !(vm.exists(name)) && (numdims > 0);
+      if (!pass)
+        error_msgs << "ERROR: loop variable already declared."
+                   << " variable name=\"" << name << "\"" << std::endl;
+      else
+        vm.add(name, base_var_decl(name, std::vector<expression>(), expr.expression_type().type()),
```

Comment:
Hi @mitzimorris , @bob-carpenter , do you have any suggestions on how to get a vector of dimension sizes out of an expression? To use `base_var_decl` to add an array element to the var map, we need to specify such a vector. I can only find how to get the number of dimensions out of an expression. Thanks!

Reply:
(It seems to me like this is not possible. Should I just declare the variable without without sizes for its dimensions? How would I do that? Presumably I want to pass a vector with length `numdims - 1`, but I'm not sure what it should contain.)

Reply:
at compile time, the size of the dimensions is unknown - you only know the number of dimensions

Reply:
Thanks, Mitzi! I am wondering then what the following constructor expects as its second argument:
`      /**
       * Construct a base variable declaration with the specified
       * name, dimensions, and base type.  The definition is set to
       * nil. 
       *
       * @param name name of variable
       * @param dims dimensions of variables
       * @param base_type base expression type of variable
       */
      base_var_decl(const std::string& name,
                    const std::vector<expression>& dims,
                    const base_expr_type& base_type);

Reply:
helper function `infer_var_dims_type` ??
(line 167 of semantic_actions_def.hpp)

Reply:
( it lists 
```     /**
       * Dimension sizes for variable.
       */
      std::vector<expression> dims_;```
)